bin/proving-service/src/proxy/status.rs

```diff
@@ -1,103 +1,257 @@
-use std::sync::Arc;
-
-use pingora::{
-    apps::HttpServerApp,
-    http::ResponseHeader,
-    prelude::*,
-    protocols::{Stream, http::ServerSession},
-    server::ShutdownWatch,
-};
-use serde::Serialize;
-use tonic::async_trait;
-use tracing::error;
+use std::{sync::Arc, time::Duration};
 
-use super::worker::WorkerHealthStatus;
-use crate::{commands::worker::ProverType, proxy::LoadBalancerState};
+use async_trait::async_trait;
+use pingora::{server::ListenFds, services::Service};
+use tokio::{net::TcpListener, sync::watch, time::interval};
+use tokio_stream::wrappers::TcpListenerStream;
+use tonic::{Request, Response, Status, transport::Server};
+use tracing::{error, info};
 
-// Status of a worker
-#[derive(Debug, Serialize)]
-pub struct WorkerStatus {
-    address: String,
-    version: String,
-    status: WorkerHealthStatus,
-}
+use super::worker::WorkerHealthStatus as RustWorkerHealthStatus;
```

Comment:
nit: Confused as to the context of the word "Rust" here. Why is the alias required?